### PR TITLE
[xcode12.2] Initial bump to Xcode 12.2 Beta 2

### PR DESCRIPTION
**Pull request by dalexsoto**

### CODE REVIEW

**Review by spouliot — APPROVED** [review]

**Review by rolfbjarne — APPROVED** [review]

File: tests/monotouch-test/MediaAccessibility/ImageCaptioningTest.cs [hunk 1]
```diff
@@ -67,6 +67,12 @@ public void GetMetadataTagPath ()
 		[Test]
 		public void SetCaption ()
 		{
+			// looks like Apple broke something inside Xcode 12.2 Beta 2 in both sim and device
+			// NSInvalidArgumentException Reason: -[__NSCFType _getValue:forType:]: unrecognized selector sent to instance
```
Comment (chamons) [inline tests/monotouch-test/MediaAccessibility/ImageCaptioningTest.cs:71]: Did you file an issue?

**Review by chamons — APPROVED** [review]

**Review by mandel-macaque — APPROVED** [review]

**Review by whitneyschmidt — APPROVED** [review]

File: tests/monotouch-test/MediaAccessibility/ImageCaptioningTest.cs (hunk 1 above)
Comment (dalexsoto) [reply tests/monotouch-test/MediaAccessibility/ImageCaptioningTest.cs:72]: Here: https://github.com/xamarin/xamarin-macios/issues/9744

**Comment by mandel-macaque** [comment]
@dalexsoto the MTouch tests can be ignored it is network related, is a known issue https://github.com/xamarin/maccore/issues/2154 , the generator ones are not

**Comment by dalexsoto** [comment]
mmm the generator failure is weird 

```
15:32:30.3844560 	tmp-test-dir/Xamarin.Tests.BGenTool.EnsureTempDir123/Test/C159.g.cs(82,20): error CS0311: The type 'QuickLook.IQLPreviewItem' cannot be used as type parameter 'T' in the generic type or method 'Runtime.GetNSObject<T>(IntPtr)'. There is no implicit reference conversion from 'QuickLook.IQLPreviewItem' to 'Foundation.NSObject'.
15:32:30.3845110 	tmp-test-dir/Xamarin.Tests.BGenTool.EnsureTempDir123/Test/M159.g.cs(82,20): error CS0311: The type 'QuickLook.IQLPreviewItem' cannot be used as type parameter 'T' in the generic type or method 'Runtime.GetNSObject<T>(IntPtr)'. There is no implicit reference conversion from 'QuickLook.IQLPreviewItem' to 'Foundation.NSObject'.
15:32:30.3845280 	tmp-test-dir/Xamarin.Tests.BGenTool.EnsureTempDir123/Test/M159.g.cs(132,21): error CS0311: The type 'QuickLook.IQLPreviewItem' cannot be used as type parameter 'T' in the generic type or method 'Runtime.GetNSObject<T>(IntPtr)'. There is no implicit reference conversion from 'QuickLook.IQLPreviewItem' to 'Foundation.NSObject'.
15:32:30.3845460 	tmp-test-dir/Xamarin.Tests.BGenTool.EnsureTempDir123/Test/M159.g.cs(134,21): error CS0311: The type 'QuickLook.IQLPreviewItem' cannot be used as type parameter 'T' in the generic type or method 'Runtime.GetNSObject<T>(IntPtr)'. There is no implicit reference conversion from 'QuickLook.IQLPreviewItem' to 'Foundation.NSObject'.
```

definition inside [tests/generator/sof20696157.cs](https://github.com/xamarin/xamarin-macios/blob/main/tests/generator/sof20696157.cs)

```csharp
[Protocol] [BaseType (typeof (NSObject))] interface M159 : IQLPreviewControllerDataSource {}
[Protocol] interface C159 : IQLPreviewControllerDataSource {}
```

The generated code is 

```csharp
[Export ("previewController:previewItemAtIndex:")]
[BindingImpl (BindingImplOptions.GeneratedCode | BindingImplOptions.Optimizable)]
[Preserve (Conditional = true)]
public IQLPreviewItem GetPreviewItem (global::QuickLook.QLPreviewController controller, nint index)
{
	if (controller == null)
		throw new ArgumentNullException (nameof (controller));
	return  Runtime.GetNSObject<IQLPreviewItem> (global::api0.Messaging.IntPtr_objc_msgSend_IntPtr_nint (this.Handle, Selector.GetHandle ("previewController:previewItemAtIndex:"), controller.Handle, index));
}

[Export ("previewController:previewItemAtIndex:")]
[BindingImpl (BindingImplOptions.GeneratedCode | BindingImplOptions.Optimizable)]
[Preserve (Conditional = true)]
public IQLPreviewItem GetPreviewItem (global::QuickLook.QLPreviewController controller, nint index)
{
	if (controller == null)
		throw new ArgumentNullException (nameof (controller));
	return  Runtime.GetNSObject<IQLPreviewItem> (global::api0.Messaging.IntPtr_objc_msgSend_IntPtr_nint (this.Handle, Selector.GetHandle ("previewController:previewItemAtIndex:"), controller.Handle, index));
}

[Export ("previewController:previewItemAtIndex:")]
[BindingImpl (BindingImplOptions.GeneratedCode | BindingImplOptions.Optimizable)]
[Preserve (Conditional = true)]
public virtual IQLPreviewItem GetPreviewItem (global::QuickLook.QLPreviewController controller, nint index)
{
	if (controller == null)
		throw new ArgumentNullException (nameof (controller));
	if (IsDirectBinding) {
		return  Runtime.GetNSObject<IQLPreviewItem> (global::api0.Messaging.IntPtr_objc_msgSend_IntPtr_nint (this.Handle, Selector.GetHandle ("previewController:previewItemAtIndex:"), controller.Handle, index));
	} else {
		return  Runtime.GetNSObject<IQLPreviewItem> (global::api0.Messaging.IntPtr_objc_msgSendSuper_IntPtr_nint (this.SuperHandle, Selector.GetHandle ("previewController:previewItemAtIndex:"), controller.Handle, index));
	}
}
```

The definition has not changed in a loooong time

https://github.com/xamarin/xamarin-macios/blob/e343e68725000041a519e243fe6f84edcbad1916/src/quicklook.cs#L91-L94

I am not really touching anything in the generator so this is even more strange

**Comment by dalexsoto** [comment]
And this did not fail in the original bump PR from @rolfbjarne for Beta 1 https://github.com/xamarin/xamarin-macios/pull/9665

**Comment by dalexsoto** [comment]
@spouliot / @rolfbjarne would you have any ideas to try here? Not sure what to try next, I mean we could disable and test later to unblock the PR but I wonder if this is hiding something more obscure.

I’ll sleep on it heh

**Comment by spouliot** [comment]
@dalexsoto I don't see any change in the PR that could lead to this failure :(
Did you compare the generated output with the one from before ? `xcode12.2` or `main`

**Comment by rolfbjarne** [comment]
I'm having a look, and it seems to be a bug in either IKVM or csc...

**Comment by spouliot** [comment]
@dalexsoto that bump ~~is~~ was a bit extreme ☹️ 

@rolfbjarne how did you plan to get those ikvm sources for net6 ? if we have to submodule later then maybe doing it would prove helpful :)

OTOH we might eventually need other fixes on top of this (or a fix already inside the changes). I think we should have master bumped to HEAD, c.c. @chamons

**Comment by dalexsoto** [comment]
@spouliot agreed, I guess in order to unblock this pr we can file an issue and keep the ball rolling for now, I think is the best we can do so we can unblock doing bindings for now.

@spouliot / @rolfbjarne / @chamons  💭s?

**Review by tj-devel709 — APPROVED** [review]

**Comment by chamons** [comment]
That sounds fine to me.

**Comment by dalexsoto** [comment]
Network related issue, https://github.com/xamarin/maccore/issues/2154 and https://github.com/xamarin/maccore/issues/2322

**Comment by rolfbjarne** [comment]
> @rolfbjarne how did you plan to get those ikvm sources for net6 ? if we have to submodule later then maybe doing it would prove helpful :)

Right now the we're using the sources in the mono archive for the net6 version of bgen. Switching to a submodule is probably the best course of action, especially now that we have another reason to do so.

**Pull request merged**